src/lib/integration_test_lib/malleable_error.ml

```diff
@@ -108,6 +108,17 @@ let contextualize context m =
   | Error hard_fail ->
       Error (Hard_fail.contextualize context hard_fail)
 
+let contextualize_soft (context : string) m =
```

Comment:
I think it might be better to write a different function, `soften_errors`, which just does the part of the logic where it downgrades all hard errors to soft errors. Then you can use this more compositionally, ie `m |> contextualize context |> soften_error`.